### DISCUSSION TITLE
How do I get the client_id of those LSP servers attached to the current buffer?

### DISCUSSION THREAD

**Discussion by nyngwang** [neovim/neovim#24105] [Q&A] [answered]
I want to check whether or not the LSP attached to the current buffer has certain capabilities, i.e. `textDocument/inlayHint`. So now I'm trying to get the `client_id` of the LSP server attached to the current buffer and using it to call `vim.lsp.get_client_by_id`. I already know that I can get the `client_id` on `LspAttach` but if the client does have the capability I will create some `autocmd`s, and thus I will create nested `autocmd`s in the callback of `LspAttach`. How to avoid nested `autocmd`s?

**Comment by clason** [comment, marked as answer]
`:h vim.lsp.get_active_clients()`

**Reply by nyngwang** [reply]
Thanks for your careful answer. I just noticed that this is the replacement of `vim.lsp.for_each_buffer_client`, which is about to be deprecated in #24104.
